src/sparse_int_set.jl

```diff
@@ -0,0 +1,258 @@
+import Base: @propagate_inbounds, zip
+
+const INT_PER_PAGE = div(ccall(:jl_getpagesize, Clong, ()), sizeof(Int))
+
+#TODO: Batch creation and allocation
+mutable struct SparseIntSet
+	packed ::Vector{Int}
+	reverse::Vector{Vector{Int}}
+	function SparseIntSet()
+		out = new(Int[], Vector{Int}[])
+		return out
+	end
+end
+
+function SparseIntSet(indices)
+	set = SparseIntSet()
+	for i in indices
+		i <= 0 && throw(ArgumentError("All Ints need to be positive."))
+		push!(set, i)
+	end
+	return set
+end
+
+SparseIntSet(i...) = SparseIntSet(i)
```

Comment:
Do other sets have this overload?

Reply:
Indeed not that I see, I changed it to mimic: [int_set.jl#L14](https://github.com/louisponet/DataStructures.jl/blob/c38959cbf3a771316b9d133be02530f47f8a942b/src/int_set.jl#L14), since all the functionality was in `union!` already anyway.